### PR TITLE
RAINCATCH-570 - update to the new browser api

### PR DESCRIPTION
**Pull request by wtrocki**
# Motivation

FileError is obsolete and was removed in most recent versions of the mobile browsers:
https://developer.mozilla.org/en-US/docs/Web/API/FileError

This fix removes error I have seen when testing sample app. I did not seen any other errors and rest of the html5 filesystem api seems to be working fine. However Lawnchair library is fairly outdated and it would be better to update it for better future browser support.

ping @david-martin @wei-lee 

More information in ticket: https://issues.jboss.org/browse/RAINCATCH-570

### CODE REVIEW

**Comment by wtrocki** [comment]
After extensive testing I found  problems with adapters version on different browsers:
screen shot 2017-02-15 at 5 14 36 pm

It would be better to Lawnchair update adapters source code (upstream seems to have much more adapters now. Despite that it looks like it's unmaintained for a year.

**Comment by wtrocki** [comment]
It seems that this PR: https://github.com/brianleroux/lawnchair/pull/228/files and recent files from the master would resolve all the problems, but it still seems that this library would not be maintained. Maybe I can investigate replacing storage with something more recent? 

@david-martin @wei-lee WDYT?

File: libs/lawnchair/lawnchairHtml5FileSystem.js [hunk 1]
```diff
@@ -1,25 +1,23 @@
...
-      case FileError.SECURITY_ERR:
+      case "SecurityError":
         msg = 'SECURITY_ERR';
         break;
-      case FileError.INVALID_MODIFICATION_ERR:
+      case "InvalidModificationError":
         msg = 'INVALID_MODIFICATION_ERR';
         break;
-      case FileError.INVALID_STATE_ERR:
+      case "InvalidStateError":
```
Comment (wei-lee) [inline libs/lawnchair/lawnchairHtml5FileSystem.js]: Will this change break backward compatibility on some of the old browsers?

Perhaps check the existence of `FileError` first?

**Comment by wei-lee** [comment]
@wtrocki I think we might want to check if `FileError` exists first to work with older version of browsers.

If we can find newer version of a library and also maintain backward compatibility, maybe we can look into it. But I am wonder what other benefits we can get from it. There is always the risk of the library may not be maintained anymore. If we are going to use a different one, I think it really should have some other benefits compared the current solution.

File: libs/lawnchair/lawnchairHtml5FileSystem.js (hunk 1 above)
Comment (wtrocki) [inline libs/lawnchair/lawnchairHtml5FileSystem.js]: I spent some time to investigate how to do it to cover most of the browsers. FileError is really old - it was depreciated couple years ago.  This code would not break anything as we just use it to log message. In previous version of the code it was throwing error and stoping entire javascript thread from processing. 
We using this only to log actual message which is carrying the same information as actual error.
I think we can even simplify this (going to add another commit)

**Comment by wtrocki** [comment]
@wei-lee Simplified entire code so `FileError` comment is not actual. In relation to changing library -  I think it's just matter of supporting recent browsers. I see that upstream has lots of PR's with fixes for browser incompatibilities that weren't merged. I couldn't find any issues with storage on my setup but it would be better (and more efficient) to just change to something that is more up to date with standards - especially when we doing extensive testing now.

**Comment by wtrocki** [comment]
ping @corinnekrych

**Pull request merged**